### PR TITLE
Implement repos

### PR DESCRIPTION
**Pull request by JeppeKorgaard98**
### Notable changes

This is the first iteration of out repositories. We are gonna look this through and apply changes and review it later on aswell.

Also resolves:
#223 
#184
#211 

### Checklist

- [x] I've run the tests and verified that nothing explodes
- [x] I've added a screenshot of the test report
- [x] The code is formatted with a C# formatter
- [x] I reviewed this pr myself

### Test report screenshot

image

### CODE REVIEW

File: ProjectBank/Infrastructure.Tests/ProjectRepositoryTests.cs [hunk 1]
```diff
@@ -4,18 +4,165 @@ public class ProjectRepositoryTests : ContextSetup, IDisposable
...
+            Role = Role.Student
+        });
+        await _projectRepository.AddUserToProjectAsync(1, 1);
+        await _projectRepository.AddUserToProjectAsync(2, 1);
+
+        var project = await _context.Projects.FindAsync(1);
 
-        var actual = _context.Projects.Find(1).Title;
+        var actual = project.Participants.ElementAt(0);
+        var expected = await _context.Users.FindAsync(1);
```
Comment (frejabv) [inline ProjectBank/Infrastructure.Tests/ProjectRepositoryTests.cs:162]: Why is the expected something we await?

File: ProjectBank/Infrastructure/ProjectRepository.cs [hunk 2]
```diff
@@ -1,5 +1,157 @@
...
+    {
+        return await _context.Projects.Where(project => project.UserId == userId).Select(project => new ProjectDTO(
+               project.Id,
+               project.Title,
+               project.Status,
+               project.UserId))
+               .ToListAsync();
+    }
+
+    public async Task UpdateProjectStatusByIdAsync(int projectId)
```
Comment (frejabv) [inline ProjectBank/Infrastructure/ProjectRepository.cs:113]: If it's only possible to set the status to closed shouldn't the method be called something like:
```suggestion
    public async Task SetProjectStatusToClosedByIdAsync(int projectId)
```

File: ProjectBank/Infrastructure/UserRepository.cs [hunk 3]
```diff
@@ -1 +1,100 @@
...
+        var project = await _context.Projects.FindAsync(projectId);
+        var listOfParticipants = new List<UserDTO>();
+        foreach (var user in project.Participants)
+        {
+            listOfParticipants.Add(new UserDTO(user.Id, user.Name));
+        }
+        return listOfParticipants;
+    }
+
+    public async Task<IReadOnlyCollection<UserDTO>> ReadUsersAsync()
```
Comment (frejabv) [inline ProjectBank/Infrastructure/UserRepository.cs:86]: Maybe this should be called:
```suggestion
    public async Task<IReadOnlyCollection<UserDTO>> ReadAllUsersAsync()
```
or
```suggestion
    public async Task<IReadOnlyCollection<UserDTO>> ReadAllAsync()
```
To correspond with the method in ProjectRepository?

File: ProjectBank/Infrastructure/ProjectRepository.cs (hunk 2 above)
Comment (JeppeKorgaard98) [reply ProjectBank/Infrastructure/ProjectRepository.cs:113]: we did this in the other branch :)

File: ProjectBank/Infrastructure.Tests/ProjectRepositoryTests.cs (hunk 1 above)
Comment (JeppeKorgaard98) [reply ProjectBank/Infrastructure.Tests/ProjectRepositoryTests.cs:162]: I think we should just run as many things async as possible hmm

File: ProjectBank/Infrastructure/UserRepository.cs (hunk 3 above)
Comment (JeppeKorgaard98) [reply ProjectBank/Infrastructure/UserRepository.cs:86]: this is done

**Review by JacobMoller — APPROVED** [review]

**Pull request merged**